src/eth/primitives/execution_conflict.rs

```diff
@@ -1,39 +1,44 @@
+use nonempty::NonEmpty;
+
 use crate::eth::primitives::Address;
 use crate::eth::primitives::Nonce;
 use crate::eth::primitives::SlotIndex;
 use crate::eth::primitives::SlotValue;
 use crate::eth::primitives::Wei;
-use crate::ext::not;
+
+#[derive(Debug)]
+pub struct ExecutionConflicts(pub NonEmpty<ExecutionConflict>);
 
 #[derive(Debug, Default)]
-pub struct ExecutionConflicts {
-    inner: Vec<ExecutionConflict>,
-}
+pub struct ExecutionConflictsBuilder(Vec<ExecutionConflict>);
 
-impl ExecutionConflicts {
+impl ExecutionConflictsBuilder {
```

Comment:
We need a builder because `ExecutionConflicts` cannot be created with an empty vec anymore.